### DISCUSSION TITLE
Noice canceling

### DISCUSSION THREAD

**Discussion by podaen** [pschatzmann/arduino-audio-tools#391] [Ideas]
I have this crazy idea  to cancel the noice that is recorded through a mic. Invert the sinewave and send it to the spreaker. I am seriously interested to try this. If it works we have a win, if it doesn't at leasted we tried.

The dogs of the neighbors are barking me out of bed when somebody is passing by. I m convinced that most of the sound comes beneath the gap of the door. And luckely that's where my spreakers are. But first I will try it with a radio and put some relaxing music on. 🙂

Do you think it this could work?

**Comment by pschatzmann** [comment]
I expect that whatever you try to do, It will  not be fast enough: you will have latency when reading the signal and add more latency when you try to output your negative signal: so instead of cancelling you might increase the noise.

I think a better approach might be that, as soon as you notice a noise level above a threshold, you start your favourite music or sound that is louder then the annoying dog!

**Comment by podaen** [comment]
That's an idea. I will give it a try. Better than a frustated dog that want to take a walk. Thanks for the advice.

**Comment by luke2023** [comment]
@podaen great idea,I suggest we can also cancel the noise after recording

**Comment by podaen** [comment]
I like some inspiration. If we could make are home a happier place to live in, we would sigh for that. Not!? Never the less it's not an easy thing to do. I think we need some equipment for that and counter the latency of the mic that is verry minimized. Until than we enjoying the sound of music. 😆

**Reply by dragoncoder047** [reply]
A good ol' LM386 would probably do the trick. If you have an analog microphone, I would just wire it to the amplifier backwards, so that the analog circuitry does the work for you. There could be a switch in the middle of this path operated by your microcontroller so that this functionality could be turned off when needed, too.

One of my cousins was setting up a sound system in his home, and he accidentally wired one of the two speakers up backwards. Then when he had the balance control set exactly in the middle, he heard nothing.
